packages/apollo-datasource-rest/src/__tests__/RESTDataSource.test.ts

```diff
@@ -89,6 +89,69 @@ describe('RESTDataSource', () => {
       expect(fetch.mock.calls[0][0].url).toEqual('https://example.com/api/foo');
     });
 
+    it('prevents parameters escaping the base host', async () => {
+      const dataSource = new (class extends RESTDataSource {
+        baseURL = 'https://api.example.com/entity';
+
+        getByUuid(entityUuid: string) {
+          return this.get(`/${entityUuid}`);
```

Comment:
We have more recently added documentation encouraging the use of `encodeURIComponent`.

We could consider explicitly supporting something like `this.get('/:entity', {entity: entityUuid})` to make encoding easier though...  Or you can just use something like the `url-assembler` package that specializes in this.